### PR TITLE
When Nice= is used, clamp to RLIMIT_NICE

### PR DESCRIPTION
**Pull request by poettering**
This is a warm-up for #11397 which was too close to being ready to leave it hanging. I did some minor updates (mostly what I pointed out in my most recent review on that PR), plus added a test case.

Since this is mostly @xnox i left his authorship of the commit intact.

/cc @xnox

### CODE REVIEW

**Comment by poettering** [comment]
Hmm, CI failure looks like a segfault in valgrind. Which appears unrelated? weird...

**Comment by mrc0mmand** [comment]
> Hmm, CI failure looks like a segfault in valgrind. Which appears unrelated? weird...

Yup, this is unfortunately a flake... I'll give it one more try

**Comment by poettering** [comment]
@mrc0mmand nice! it passed this time! thanks!

File: src/test/test-process-util.c [hunk 1]
```diff
@@ -601,6 +604,92 @@ static void test_ioprio_class_from_to_string(void) {
...
+        assert_se(r >= 0);
+
+        if (r == 0) {
+                bool full_test;
+                int p, q;
+                /* child */
+
+                /* rlimit of 30 equals nice level of -10 */
+                if (setrlimit(RLIMIT_NICE, &RLIMIT_MAKE_CONST(30)) < 0) {
+                        /* If this fails we are probably unprivielged or in a userns of some kind, let's skip
```
Comment (keszybz) [inline src/test/test-process-util.c:620]: typo here

**Pull request merged**